src/page/EditCheck/EditCheck.tsx

```diff
@@ -18,9 +19,16 @@ const EditCheckContent = () => {
   const { data: checks, isError, isLoading, error, refetch } = useChecks();
   const check = checks?.find((c) => c.id === Number(id));
 
+  const { data: alertsPerCheck } = useListAlertsForCheck(check?.id);
```

Comment:
We could leave this hook call here to increase perceived performance so we fetch the data before reaching the alerting step (maybe encapsulate it in another function with a clear name or leave a comment explaining why) but we don't have to pass down the response.

Just call `useListAlertsForCheck` from the `<AlertsPerCheck />` component because we are going to want the `isLoading` and `isError` states as well as probably the `refetch` method, etc. and render the correct UI accordingly.

When I tested this the dev environment wasn't deployed so naturally I got a 404 but was clueless as a user I had hit an error. We need to decide what we want to present in the UI if this endpoint isn't available in production.